### PR TITLE
Add option to install CRD as a part of chart install in flyte-binary

### PR DESCRIPTION
**Pull request by marrrcin**
## Why are the changes needed?
Due to strict RBAC policies in some k8s clusters, it's impossible to create `ClusterRole` with wide set of permissions for service account used in `flyte-binary` deployment. A namespace-scoped `Role` could be used, but it requires:
* creating own `Role` and `RoleBinding` (possible by creating a sub-chart)
* disabling installation of `FlyteWorkflow` CRD at runtime, after Helm install

More context: https://flyte-org.slack.com/archives/C01P3B761A6/p1729865751682139

## What changes were proposed in this pull request?

1. Added CRD for FlyteWorkflow into flyte-binary chart
2. Added flag in values to configure propeller to skip CRD installation
3. Added flag in values to enable CRD installation at `helm install`

## How was this patch tested?

1. Created new k3d cluster
5. Installed modified Helm chart 2 times:
3.1. with `helm install`-level CRDs creation enabled and propeller-level CRDs creation disabled.
2.2. with `helm install`-level CRDs creation disabled and propeller-level CRDs creation enabled.
6. Both cases worked.

### Setup process
See above.

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Related PRs

N/A

## Docs link

N/A

### CODE REVIEW

**Comment by chrisrun** [comment]
+1

File: charts/flyte-binary/values.yaml [hunk 1]
```diff
@@ -375,6 +378,10 @@ rbac:
   # extraRules Add additional rules to the ClusterRole
   extraRules: []
 
+crds:
```
Comment (davidmirror-ops) [inline charts/flyte-binary/values.yaml:381]: Not sure I follow, is this section still needed? Similar to how flyte-core handles this, we could leave the `configuration.propeller.createCrd` key as both condition for the flyteworkflow CRD Helm template and as the input for the `create-flyteworkflow-crd` key in the configmap.

Comment (marrrcin) [reply charts/flyte-binary/values.yaml]: Changed.
The original intent was to be able to get rid of CRDs installation completely, but that might be too confusing.

File: charts/flyte-binary/templates/crds/flyteworkflow.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{{- if not .Values.configuration.propeller.createCRDs }}
```
Comment (davidmirror-ops) [inline charts/flyte-binary/templates/crds/flyteworkflow.yaml:1]:
```suggestion
{{- if  .Values.configuration.propeller.createCRDs }}
```
I think this should be the logic?

**Review by davidmirror-ops — CHANGES_REQUESTED** [review]

**Comment by marrrcin** [comment]
@davidmirror-ops pushed! :)

File: charts/flyte-binary/templates/crds/flyteworkflow.yaml (hunk 2 above)
Comment (marrrcin) [reply charts/flyte-binary/templates/crds/flyteworkflow.yaml:1]: No, the logic is correct. If `propeller` is set to create CRDs at runtime, then we don't want to install them as a part of helm chart installation.

Comment (davidmirror-ops) [reply charts/flyte-binary/templates/crds/flyteworkflow.yaml:1]: You're right. Thanks for your patience, this helped me understand better how the CRD install operation works.

**Review by davidmirror-ops — APPROVED** [review]
I just installed flyte-binary using this chart and can confirm the expected behavior about the CRD.
Thank you!

**Comment by marrrcin** [comment]
Awesome, thanks!

**Pull request merged**